The solution of the "Website post" problem

sargisEg opened the pull request.

sargisEg committed:
The solution of the "Website post" problem

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/entity/Post.java:
```diff
@@ -0,0 +1,105 @@
+    private String description;
+
+    @ManyToOne
+    @JoinColumn(name = "user_id", nullable = false)
```
foreign key constraint is missing

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/entity/Post.java:
```diff
@@ -0,0 +1,105 @@
+    @JoinColumn(name = "user_id", nullable = false)
+    private User user;
+
+    @ManyToMany(fetch = FetchType.EAGER)
```
plase do not mark as eager

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/entity/Post.java:
```diff
@@ -0,0 +1,105 @@
+
+    @ManyToMany(fetch = FetchType.EAGER)
+    @JoinColumn(name = "images", nullable = false)
+    private List<Image> images;
```
I even suggest to remove this for now

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/facade/post/GetAllUserPostsResponseDto.java:
```diff
@@ -0,0 +1,51 @@
+public final class GetAllUserPostsResponseDto {
+
+    private final User user;
+    private final List<Post> posts;
```
please do not use entity in dto class

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/facade/post/PostCreateRequestDto.java:
```diff
@@ -0,0 +1,66 @@
+    private final String username;
+    private final String title;
+    private final String description;
+    private final List<Image> images;
```
please do not use entity in dto class

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/facade/post/PostCreateResponseDto.java:
```diff
@@ -0,0 +1,75 @@
+    private final User user;
+    private final String title;
+    private final String description;
+    private final List<Image> images;
```
please do not use entity in dto class

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/facade/post/PostCreateResponseDto.java:
```diff
@@ -0,0 +1,75 @@
+public final class PostCreateResponseDto {
+
+    private final Long postId;
+    private final User user;
```
please do not use entity in dto class

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/facade/post/PostCreateRequestDto.java:
```diff
@@ -0,0 +1,66 @@
+        Assert.hasText(username, "Username should not be null or empty");
+        Assert.hasText(title, "Title should not be null or empty");
+        Assert.hasText(description, "Description should not be null or empty");
+        Assert.notNull(images, "Images should not be null");
```
this can be null

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/facade/post/PostFacade.java:
```diff
@@ -0,0 +1,10 @@
+
+    PostCreateResponseDto create(PostCreateRequestDto dto);
+
+    UploadImageResponseDto uploadImage(UploadImageRequest dto);
```
Please change UploadImageRequest to UploadPostImageRequestDto

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/facade/post/PostFacade.java:
```diff
@@ -0,0 +1,10 @@
+
+    UploadImageResponseDto uploadImage(UploadImageRequest dto);
+
+    GetAllUserPostsResponseDto getAllUserPosts(GetAllUserPostsRequestDto dto);
```
this can take username only

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/facade/post/PostFacadeImpl.java:
```diff
@@ -0,0 +1,121 @@
+
+        final User user = userService.findByUsername(dto.getUsername())
+                .orElseThrow(() -> {
+                    throw new UserNotFoundException(dto.getUsername());
```
please return an error in facade instead of throwing an exception

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/facade/post/PostFacadeImpl.java:
```diff
@@ -0,0 +1,121 @@
+        final Optional<Image> optionalImage = imageService.findByBlobId(dto.getBlobId());
+
+        if (optionalImage.isPresent()) {
+            throw new ImageAlreadyExistException(dto.getBlobId());
```
please do not throw exception in facade

gevorgyanarman commented on src/main/java/com/aca/homework/week18/website/service/core/UserService.java:
```diff
@@ -0,0 +1,17 @@
+
+import java.util.Optional;
+
+@Service
```
please sen this annotation on impl calsses

sargisEg committed:
Added test for PostFacadeImpl and modified UserFacadeTest,
Removed List of Images from Post entity,
Added postId field for Image entity,
Added mappers for UserDto, PostDto and ImageDto,
Removed exceptions from UserFacadeImpl and PostFacadeImpl classes
